local-volume/provisioner/pkg/deleter/deleter.go

```diff
@@ -129,12 +130,13 @@ func (d *Deleter) asyncDeletePV(pv *v1.PersistentVolume, volType string, mountPa
 
 	// Remove API object
 	if err := d.APIUtil.DeletePV(pv.Name); err != nil {
-		// TODO: Does delete return an error if object has already been deleted?
-		deletingLocalPVErr := fmt.Errorf("Error deleting PV %q: %v", pv.Name, err.Error())
-		d.RuntimeConfig.Recorder.Eventf(pv, v1.EventTypeWarning, common.EventVolumeFailedDelete,
-			deletingLocalPVErr.Error())
-		glog.Error(deletingLocalPVErr)
-		return
+		if !errors.IsNotFound(err) {
```

Comment:
Are there any other types of errors (besides IsNotFound) that could occur? 

If yes, should we continue for those errors or log an error/event & return?

Reply:
No, I try with kubectl delete and find kube-apiserver just retuen 404.

Reply:
If this is the only type of error ever returned by DeletePV, I think that we can remove the if statement check for IsNotFound, aka `if !errors.IsNotFound(err) {`.

Reply:
I mean if no pv exists, delete will only fail with `NotFound error`, there are still other cases like `timeout`, etc. So This if statement should not be removed.

Reply:
Can you add a unit test for this?

Reply:
@msau42 add a ut case for deleting non-exist pv